recipe/meta.yaml

```diff
@@ -16,7 +16,7 @@ build:
 
 requirements:
   build:
-    - toolchain
+    - {{ compiler("c") }}
     - boost-cpp 1.66.0
```

Comment:
thanks @msarahan! The reason I pinned `boost-cpp`is that, for some reason, a different version was used in the test section, leading to the executable not finding the library.

>  Think build = build tools, host = stuff to include/link against.

so you mean that, in this specific case, I should likely leave only `automake` and `autoconf` (and the compiler directive, of course) in the `build` section, and move the rest to a `host` section? 

I'm a novice here with the desire to learn, so please be patient if I ask dumb questions...

Reply:
Pinning values come from here: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L210 - because there is a name match, you don't have to pin manually.

Because boost-cpp does not have run_exports yet, and because it is not one of the ones in the global pin_run_as_build collection (https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L49), you should put this in your run dependencies instead of boost-cpp:

``{{ pin_compatible('boost-cpp', max_pin='x.x.x') }}``

This says "pin compatible to whatever boost version is installed in host at build time.  Do so to 3 version values."  This will take care of the runtime version.

More info at https://conda.io/docs/user-guide/tasks/build-packages/variants.html#customizing-compatibility

Reply:
that's invaluable (and increadibly clear) info, thanks! I'll try in a minute.